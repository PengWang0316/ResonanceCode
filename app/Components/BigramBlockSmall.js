import React from 'react';
import PropTypes from 'prop-types';

import styles from '../styles/BigramBlockSmall.module.css';
import taijiSmall from '../imgs/taijiSmall.png';

const BigramBlockSmall = ({ lineText, position }) => (
  <div className="mr-4">
    <img className={styles.taijiPic} src={taijiSmall} alt="taiji" />
    <div className="d-flex">
      <div className={`${styles.bigramBlock} ${styles.blockBottomBorder}  ${styles.blockRightBorder} ${position === 1 ? styles.selected : ''}`} /><div className={`${styles.bigramBlock} ${styles.blockBottomBorder} ${position === 0 ? styles.selected : ''}`} />
    </div>
    <div className="d-flex">
      <div className={`${styles.bigramBlock} ${styles.blockRightBorder} ${position === 2 ? styles.selected : ''}`} /><div className={`${styles.bigramBlock} ${position === 3 ? styles.selected : ''}`} />
    </div>
    <div className={styles.lineText}>{lineText}</div>
  </div>
);
BigramBlockSmall.propTypes = {
  lineText: PropTypes.string.isRequired,
  position: PropTypes.number.isRequired
};
export default BigramBlockSmall;
