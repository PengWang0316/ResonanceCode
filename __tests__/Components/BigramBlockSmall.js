import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import BigramBlockSmall from '../../app/Components/BigramBlockSmall';

describe('BigramBlockSmall test', () => {
  const defaultProps = {
    lineText: 'test line Text',
    position: 0
  };
  // const getShallowComponent = (props = defaultProps) => shallow(<BigramBlockSmall {...props} />);

  test('BigramBlockSmall position 0 snapshot', () => expect(renderer.create(<BigramBlockSmall {...defaultProps} />).toJSON()).toMatchSnapshot());

  test('BigramBlockSmall position 1 snapshot', () => expect(renderer.create(<BigramBlockSmall {...{ ...defaultProps, position: 1 }} />).toJSON()).toMatchSnapshot());

  test('BigramBlockSmall position 2 snapshot', () => expect(renderer.create(<BigramBlockSmall {...{ ...defaultProps, position: 2 }} />).toJSON()).toMatchSnapshot());

  test('BigramBlockSmall position 3 snapshot', () => expect(renderer.create(<BigramBlockSmall {...{ ...defaultProps, position: 3 }} />).toJSON()).toMatchSnapshot());
});
