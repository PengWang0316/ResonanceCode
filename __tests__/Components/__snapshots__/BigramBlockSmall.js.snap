// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigramBlockSmall test BigramBlockSmall position 0 snapshot 1`] = `
<div
  className="mr-4"
>
  <div
    className="d-flex"
  >
    <div
      className="bigramBlock blockBottomBorder  blockRightBorder "
    />
    <div
      className="bigramBlock blockBottomBorder conception"
    />
  </div>
  <div
    className="d-flex"
  >
    <div
      className="bigramBlock blockRightBorder "
    />
    <div
      className="bigramBlock "
    />
  </div>
  <div
    className="lineText"
  >
    test line Text
  </div>
</div>
`;

exports[`BigramBlockSmall test BigramBlockSmall position 1 snapshot 1`] = `
<div
  className="mr-4"
>
  <div
    className="d-flex"
  >
    <div
      className="bigramBlock blockBottomBorder  blockRightBorder growth"
    />
    <div
      className="bigramBlock blockBottomBorder "
    />
  </div>
  <div
    className="d-flex"
  >
    <div
      className="bigramBlock blockRightBorder "
    />
    <div
      className="bigramBlock "
    />
  </div>
  <div
    className="lineText"
  >
    test line Text
  </div>
</div>
`;

exports[`BigramBlockSmall test BigramBlockSmall position 2 snapshot 1`] = `
<div
  className="mr-4"
>
  <div
    className="d-flex"
  >
    <div
      className="bigramBlock blockBottomBorder  blockRightBorder "
    />
    <div
      className="bigramBlock blockBottomBorder "
    />
  </div>
  <div
    className="d-flex"
  >
    <div
      className="bigramBlock blockRightBorder maturation"
    />
    <div
      className="bigramBlock "
    />
  </div>
  <div
    className="lineText"
  >
    test line Text
  </div>
</div>
`;

exports[`BigramBlockSmall test BigramBlockSmall position 3 snapshot 1`] = `
<div
  className="mr-4"
>
  <div
    className="d-flex"
  >
    <div
      className="bigramBlock blockBottomBorder  blockRightBorder "
    />
    <div
      className="bigramBlock blockBottomBorder "
    />
  </div>
  <div
    className="d-flex"
  >
    <div
      className="bigramBlock blockRightBorder "
    />
    <div
      className="bigramBlock seeding"
    />
  </div>
  <div
    className="lineText"
  >
    test line Text
  </div>
</div>
`;
